### DISCUSSION TITLE
What are the specific use cases for `tauri-plugin-persisted-scope`?

### DISCUSSION THREAD

**Discussion by percy507** [tauri-apps/tauri#8540] [Q&A] [answered]
https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/persisted-scope

**Comment by FabianLars** [comment, marked as answer]
It saves scope configs added via these apis https://docs.rs/tauri/latest/tauri/scope/struct.FsScope.html, the built-in dialog, and file drops (if tauri's file drop is enabled) to disk so they are preserved across app restarts.

Example: Imagine you're working on a code editor, your user clicks on "Open Project" and selects their project in the file picker dialog. Tauri will then extend the fs scope with the selected path automatically so you can immediatly use the fs javascript apis.
Now without the plugin, users would have to click on "Open Project" again after restarting the app. With the plugin you could add a session restore feature (== the last used project is opened automatically) without having to worry about scopes yourself.

**Reply by percy507** [reply]
Thanks. Very clear.

**Reply by haexhub** [reply]
sorry to resurrect this old topic, but could you please explain how i can reopen a file that was opened via persisted-scope?
The fs plugin always wants to have a BaseDir. But what if the file is outside of this?

**Reply by FabianLars** [reply]
> The fs plugin always wants to have a BaseDir. 

This is not true. The `dir` option is optional. The fs apis accept absolute paths without the option as well.

**Reply by haexhub** [reply]
thanks for your response and the clarification. I thought the BaseDir is mandatory, as all examples in the documentation I found where using it. 
Thank you very much!

**Reply by RareParsec** [reply]
I think the contents of this answer would be very useful to have in the persisted-scope docs itself (https://v2.tauri.app/plugin/persisted-scope/).

Happy to open a tiny PR if needed.

**Reply by FabianLars** [reply]
prs always appreciated :)
